src/EventStore.LogV3/RecordCreator.cs

```diff
@@ -1,9 +1,30 @@
 ﻿using System;
+using System.Text;
+using System.Text.Unicode;
 using EventStore.LogCommon;
 
 namespace EventStore.LogV3 {
 	// Nice-to-use wrappers for creating and populating the raw structures.
 	public static class RecordCreator {
+		// these exist to make the code more obviously correct
+		private static byte CurrentVersion(this ref Raw.EpochHeader _) => LogRecordVersion.LogRecordV1;
+		private static byte CurrentVersion(this ref Raw.PartitionTypeHeader _) => LogRecordVersion.LogRecordV0;
+		private static byte CurrentVersion(this ref Raw.StreamTypeHeader _) => LogRecordVersion.LogRecordV0;
+
+		private static LogRecordType Type(this ref Raw.EpochHeader _) => LogRecordType.System;
+		private static LogRecordType Type(this ref Raw.PartitionTypeHeader _) => LogRecordType.PartitionType;
+		private static LogRecordType Type(this ref Raw.StreamTypeHeader _) => LogRecordType.StreamType;
+		const int MaxStringBytes = 100;
+		static void PopulateString(string str, Memory<byte> target) {
+			Utf8.FromUtf16(str, target.Span, out _, out var bytesWritten, true, true);
+			if (bytesWritten > MaxStringBytes)
+				throw new ArgumentException($"Name \"{str}s\"is longer than 100 bytes");
```

Comment:
```suggestion
				throw new ArgumentException($"Name \"{str}\" is longer than 100 bytes");
```

Looks like a small typo and a space is missing

Reply:
yep good spot!